packages/legend-application-studio/src/components/editor/edit-panel/service-editor/ServiceExecutionQueryEditor.tsx

```diff
@@ -332,8 +341,9 @@ export const ServiceExecutionQueryEditor = observer(
               }),
             );
             if (
-              openQueryBuilderTextEditor &&
-              !embeddedQueryBuilderState.queryBuilderState?.isQuerySupported
+              !editQueryInFormMode ||
```

Comment:
why is this logic here?
Should be part of the setup function
`setupQueryBuilderState`. 
where we are setting up the queryBuilderState. 
This would prevent unnecessary `?` and put all the setup for the queryBuilderState in one place